tests/unit/test_zos_files.py

```diff
@@ -102,3 +102,21 @@ def test_rename_dataset_member_raises_exception(self):
             Files(self.test_profile).rename_dataset_member("MY.DS.NAME", "MEMBER1", "MEMBER1N", "RANDOM")
 
         self.assertEqual(str(e_info.exception), "Invalid value. Valid options are SHRW or EXCLU.")
+
+    @mock.patch('requests.Session.send')
+    def test_rename_dataset_sends_specific_request(self, mock_send_request):
+        """Test renaming a data set sends the expected request"""
+        mock_send_request = Files(self.test_profile)
+        mock_send_request.rename_dataset = mock.Mock()
```

Comment:
BTW, this seems weird, maybe only my lack of experience with `mock` but I don't see other tests overriding the member that you later call, they rather set the `return_value` instead. How is this going to call the tested method?

Reply:
Not experienced enough either.
return_value is what you get as a result when you call the mocked method. So if you call it with some args and set return_value it returns what you set. This is in case you want to check/use what the method returns.
Here I'm using only the args passed to check if they are indeed what the method used, so I omit setting return_value, because I won't need or use that.